build_tools/build_mlir.sh

```diff
@@ -40,6 +40,8 @@ CMAKE_FLAGS+=" -DLLVM_ENABLE_EH=ON"
 
 cmake -G Ninja "$LLVM_REPO_DIR"/llvm $CMAKE_FLAGS
 
-ninja
+JOBS=$(nproc 2>/dev/null || sysctl -n hw.ncpu)
+ninja -j$JOBS
```

Comment:
Ninja defaults to all CPU cores available, this is not `make`.

Reply:
> Ninja defaults to all CPU cores available, this is not `make`.

I heard it is not the maximum, I don't know if I manually set it to available, which could make a difference. Let me double-check it.

Reply:
I heard that the difference is probably available or preemptive. Probably, when other jobs are running, this can grab resources, just a guess. But, you are right, it should be the maximum cores available.

Reply:
Yeah, on systems with constrained resources ninja's approach can lead to annoying lockups. It can be advisable to use one core less than the max. But your change is functionally equivalent right now.

Reply:
@fruffy It is more interesting actually. On my MBP laptop I'm having total 14 cores, but among those 10 performance and 4 efficiency. So I can go `-j 14`, but in the reality it does not make sense to go beyond `-j 10` as I really see difference in build speed in performance vs efficiency cores. It's better to keep them for "normal" tasks.

Reply:
Actually, I checked that the Ubuntu runner only has 2 cores. :0

```
2 cores for standard GitHub-hosted runners
Processor: x86_64
Memory: Approximately 7 GB of RAM
```
We can add those to the action workflow.
```
- name: Detailed System Information
  run: lscpu
```

Reply:
This is public information: https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

No need to guess

Reply:
> Actually, I checked that the GitHub action Ubuntu runner only has 2 cores. :0

They should have four cores, 2 if you are using a private repository.